pennylane/templates/subroutines/amplitude_amplification.py

```diff
@@ -36,10 +36,10 @@ def _get_fixed_point_angles(iters, p_min):
     gamma = np.cos(np.arccos(1 / delta, dtype=np.complex128) / iters, dtype=np.complex128) ** -1
 
     alphas = [
-        2 * np.arctan(1 / (np.tan(2 * np.pi * j / iters) * np.sqrt(1 - gamma**2)))
+        float(2 * np.arctan(1 / (np.tan(2 * np.pi * j / iters) * np.sqrt(1 - gamma**2))))
```

Comment:
Why the original implementation causes qjit issues? Is it because of a bug/issue elsewhere?

Reply:
Catalyst uses a safe casting function that allows upcasting (integers can become floats), but disallows downcasting (complex cannot become a float).

Reply:
Then we might add a test checking that the alphas are always float, if they have to be.

Reply:
I added the tests